subsys/fs/littlefs_fs.c

```diff
@@ -646,31 +658,38 @@ static int littlefs_flash_init(struct fs_littlefs *fs, uintptr_t dev_id)
 	fs->backend = (void *) *fap;
 	return 0;
 }
+#endif /* CONFIG_FS_LITTLEFS_FMP_DEV */
 
 static int littlefs_init_backend(struct fs_littlefs *fs, uintptr_t dev_id, int flags)
 {
 	int ret = 0;
 
+	__ASSERT_NO_MSG((IS_ENABLED(CONFIG_FS_LITTLEFS_FMP_DEV) && !littlefs_on_blkdev(flags)) ||
```

Comment:
No. If somebody makes mistake the device can get in  brick loop, when attempting to mount system.
I know that this should be catch in tests, but if tests fail to discover the probelI know that this is impossible situation, or will happen once and gets fixed, but I would rather have -ENOTSUP reported in such situation.

Better make it so that if flags are not correctly selected the init returns with error but allows device to work, so that at least it could be updated or have logs pulled remotely,

Reply:
I know that there are __ASSERTS used in other parts of LittleFS code, but I would avoid it as much as possible.

Reply:
done